Automattic/wp-calypso
# media: migrate `deleteMedia` to use React Query

This PR is the first in a series of PRs to migrate the media library to React Query. I'm going to tackle mutations first and integrate them intermediately with state which currently still is in Redux. That way we'll get rid of all the Redux related code step by step similar to how we migrated the media library from Flux to Redux.

#### Changes proposed in this Pull Request

* Media: migrate `deleteMedia` to use React Query
* Remove Redux action creators, thunks and data layer parts related to `deleteMedia`

#### Testing instructions

* Go to `/media`
* Select one or more items and attempt do delete them; items should be removed from the list (you'll see a network request for each selected item)
* Block network requests to `/sites/*/media/*/delete`
* Attempt another delete, you now should see an error message
* Unblock network requests and attempt to delete again, the notice should be removed and the item removed from the list

Open the editor and do all of the steps above as well from the media modal. Additionally,

* Create an image block and select a media item
* Hit _Replace_ and delete the previously selected media item
* The block should be updated upon deletion

Related to #58574
